dist/src/main/java/io/camunda/application/commons/backup/BackupPriorityConfiguration.java

```diff
@@ -192,4 +201,83 @@ public BackupPriorities backupPriorities() {
     LOG.debug("Prio6 are {}", prio6);
     return new BackupPriorities(prio1, prio2, prio3, prio4, prio5, prio6);
   }
+
+  private boolean getIsElasticsearch() {
+    final Optional<Boolean> result =
+        allMatch(
+            Optional::empty,
+            differentConfigFor("database.type"),
+            Map.of(
+                "operate",
+                Optional.ofNullable(operateProperties)
+                    .map(ignored -> DatabaseInfo.isCurrent(DatabaseType.Elasticsearch)),
+                "tasklist",
+                Optional.ofNullable(tasklistProperties)
+                    .map(prop -> prop.getDatabase().equals(TasklistProperties.ELASTIC_SEARCH)),
+                "optimize",
+                Optional.ofNullable(optimizeIsElasticSearch)),
+            skipEmptyOptional());
+    if (result.isEmpty()) {
+      throw new IllegalArgumentException(NO_CONFIG_ERROR_MESSAGE);
+    }
+    return result.get();
+  }
+
+  private List<Prio6Backup> getPrio6Backups(
+      final Boolean isElasticsearch, final OptimizeIndexNameService indexService) {
+    final List<Prio6Backup> prio6 = new ArrayList<>();
+    {
+      final var indices =
+          isElasticsearch
+              ? ElasticSearchSchemaManager.getAllNonDynamicMappings()
+              : OpenSearchSchemaManager.getAllNonDynamicMappings();
+      if (isElasticsearch) {
+        for (final IndexMappingCreator<?> index : indices) {
+          if (index instanceof final Prio6Backup p) {
+            prio6.add(
+                new OptimizePrio6Delegate<>(
+                    (IndexMappingCreator<?> & Prio6Backup) p, indexService));
+          }
+        }
+      }
+    }
+    return prio6;
+  }
+
+  private String getIndexPrefix() {
+    final var indexOptional =
+        allMatch(
+            Optional::empty,
+            differentConfigFor("indexPrefix"),
+            Map.of(
+                "operate",
+                Optional.ofNullable(operateProperties).map(OperateProperties::getIndexPrefix),
+                "tasklist",
+                Optional.ofNullable(tasklistProperties).map(TasklistProperties::getIndexPrefix),
+                "optimize",
+                Optional.ofNullable(optimizePrefix)),
```

Comment:
This might not be right thing to do as the optimizePrefix has "optimize" as default value, in contrast with the other apps which have "" as default. The other application preprend the component name to all indices (it's not part of the config)